core/src/main/java/feign/Request.java

```diff
@@ -128,7 +128,7 @@ public static Request create(HttpMethod httpMethod,
     this.httpMethod = checkNotNull(method, "httpMethod of %s", method.name());
     this.url = checkNotNull(url, "url");
     this.headers = checkNotNull(headers, "headers of %s %s", method, url);
-    this.body = body;
+    this.body = (body != null) ? body : Body.EMPTY;
```

Comment:
That is going to be a problem...
https://github.com/OpenFeign/feign/blob/master/httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java#L135

There are clients out there that change behavior when body is null

Reply:
Hmmm this is strange. Tests run without any issues. Ok, I have another suggestion. We can just return empty byte array when call Request::body and keep Body null. But I think this is not a good solution

Reply:
The reason the tests run without any issues mean that the case is probably not covered with them. The erroneous behavior would appear in runtime.